File: app/src/main/java/com/veganbeauty/app/features/shop/product/detail/ShopReviewAdapter.java
package com.veganbeauty.app.features.shop.product.detail;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.veganbeauty.app.databinding.ShopItemReviewBinding;

import java.util.List;

public class ShopReviewAdapter extends RecyclerView.Adapter<ShopReviewAdapter.ViewHolder> {

    private List<ProductReview> items;

    public ShopReviewAdapter(List<ProductReview> items) {
        this.items = items;
    }

    public void updateData(List<ProductReview> newItems) {
        this.items = newItems;
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        ShopItemReviewBinding binding = ShopItemReviewBinding.inflate(
                LayoutInflater.from(parent.getContext()), parent, false
        );
        return new ViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.bind(items.get(position));
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        private final ShopItemReviewBinding binding;

        public ViewHolder(ShopItemReviewBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }

        public void bind(ProductReview item) {
            binding.tvReviewerName.setText(item.getReviewerName());
            
            String avatarChar = "U";
            if (item.getReviewerName() != null && !item.getReviewerName().isEmpty()) {
                avatarChar = item.getReviewerName().substring(0, 1).toUpperCase();
            }
            binding.tvAvatarChar.setText(avatarChar);
            binding.tvReviewComment.setText("\"" + item.getComment() + "\"");

            ImageView[] starImageViews = {
                binding.ivStar1,
                binding.ivStar2,
                binding.ivStar3,
                binding.ivStar4,
                binding.ivStar5
            };

            for (int i = 0; i < starImageViews.length; i++) {
                if (i < item.getRating()) {
                    starImageViews[i].setVisibility(View.VISIBLE);
                } else {
                    starImageViews[i].setVisibility(View.GONE);
                }
            }
        }
    }
}

File: app/src/main/java/com/veganbeauty/app/features/shop/product/detail/ShopReviewAdapter.kt
package com.veganbeauty.app.features.shop.product.detail

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.veganbeauty.app.databinding.ShopItemReviewBinding

class ShopReviewAdapter(
    private var items: List<ProductReview>
) : RecyclerView.Adapter<ShopReviewAdapter.ViewHolder>() {

    fun updateData(newItems: List<ProductReview>) {
        items = newItems
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val binding = ShopItemReviewBinding.inflate(
            LayoutInflater.from(parent.context), parent, false
        )
        return ViewHolder(binding)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bind(items[position])
    }

    override fun getItemCount(): Int = items.size

    class ViewHolder(private val binding: ShopItemReviewBinding) : RecyclerView.ViewHolder(binding.root) {
        fun bind(item: ProductReview) {
            binding.tvReviewerName.text = item.reviewerName
            binding.tvAvatarChar.text = if (item.reviewerName.isNotEmpty()) {
                item.reviewerName.take(1).uppercase()
            } else {
                "U"
            }
            binding.tvReviewComment.text = "\"${item.comment}\""

            // Set star rating
            val starImageViews = listOf(
                binding.ivStar1, 
                binding.ivStar2, 
                binding.ivStar3, 
                binding.ivStar4, 
                binding.ivStar5
            )
            starImageViews.forEachIndexed { index, imageView ->
                if (index < item.rating) {
                    imageView.visibility = View.VISIBLE
                } else {
                    imageView.visibility = View.GONE
                }
            }
        }
    }
}
